backend_server.cc

```diff
@@ -0,0 +1,78 @@
+#include "backend_server.h"
+#include "key_value_store.h"
+
+#include <map>
+#include <string>
+#include <utility>
+#include <optional>
+
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/server.h>
+#include <grpcpp/server_builder.h>
+#include <grpcpp/server_context.h>
+#include <grpcpp/security/server_credentials.h>
+
+#include "key_value_store.grpc.pb.h"
+
+grpc::Status KeyValueStoreImpl::put(grpc::ServerContext *context, const chirp::PutRequest *request, chirp::PutReply *reply) {
+  if (request == nullptr) {
```

Comment:
Do you ever pass a nullptr to request? Is this even possible in your code?

Reply:
I never do, but I think barath recommended it as an extra sanity check